service_capacity_modeling/models/org/netflix/ddb.py

```diff
@@ -0,0 +1,584 @@
+import math
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from pydantic import BaseModel
+from pydantic import Field
+
+from service_capacity_modeling.interface import AccessConsistency, certain_float
+from service_capacity_modeling.interface import AccessPattern
+from service_capacity_modeling.interface import CapacityDesires
+from service_capacity_modeling.interface import CapacityPlan
+from service_capacity_modeling.interface import CapacityRequirement
+from service_capacity_modeling.interface import certain_int
+from service_capacity_modeling.interface import Clusters
+from service_capacity_modeling.interface import Consistency
+from service_capacity_modeling.interface import DataShape
+from service_capacity_modeling.interface import Drive
+from service_capacity_modeling.interface import FixedInterval
+from service_capacity_modeling.interface import GlobalConsistency
+from service_capacity_modeling.interface import Instance
+from service_capacity_modeling.interface import Interval
+from service_capacity_modeling.interface import QueryPattern
+from service_capacity_modeling.interface import RegionContext
+from service_capacity_modeling.interface import Requirements
+from service_capacity_modeling.interface import Service
+from service_capacity_modeling.interface import ServiceCapacity
+from service_capacity_modeling.models import CapacityModel
+
+
+class NflxDynamoDBArguments(BaseModel):
+    number_of_regions: int = Field(
+        default=1,
+        description="How many regions the dynamoDB table needs to be created. "
+        "This determines if we should use wcu's "
+        "or rwcu's and to correctly estimate the transfer costs",
+    )
+    transactional_write_percent: float = Field(
+        default=0,
+        description="Coordinated writes, if multiple items in single write "
+        "then all-or-nothing guarantee for writes. "
+        "Default is zero, assuming most writes are single item puts",
+    )
+    transactional_read_percent: float = Field(
+        default=0,
+        description="Coordinated reads, if multiple items in "
+        "single read then all-or-nothing guarantee for read. "
+        "Default is zero",
+    )
+    estimated_mean_item_size_bytes: int = Field(
```

Comment:
I'm a little confused why we need this argument when we have the write size argument but :shrug:

Reply:
There are currently two arguments `estimated_mean_read_size_bytes` and `estimated_mean_write_size_bytes` and I wasn't sure which represents the item_size. So to disambiguate, introduced as extra_model_argument.

Reply:
mean_write_size_bytes is the item size.

Reads is separate because most read apis can return _multiple_ items, so reads tend to be larger than writes.